### DISCUSSION TITLE
Running a background process in Action stopped working

### DISCUSSION THREAD

**Discussion by LarsKemmann** [community/community#25743] [Actions]
<p>I’ve described the issue here and linked to the Action run logs (public open-source project). Sadly, I can only include two links in a post as a new user.<br>
<a href="https://github.com/Azure/Azurite/issues/1271" rel="noopener nofollow ugc">Azurite no longer runs correctly as a background task in GitHub Actions · Issue #1271 · Azure/Azurite</a></p>
<p>I’ve noticed that, since today, Azurite (a tool I use for emulating Azure Storage in automated tests &amp; local development) fails to start correctly in my GitHub Actions pipeline and instead dumps some unknown text/binary content to the GitHub Action log.</p>
<p>For logs - see post above (due to new-user link limit).</p>
<p><a class="mention" href="/ethomson">@ethomson</a> (or other awesome GitHub team members) Can you take a look at why this might have suddenly stopped working? It looks like all the versions/dependencies in my Action are the same, so the only thing that seemingly has changed might be something with the underlying (Actions) platform. I originally followed <a href="https://github.com/Azure/Azurite/issues/451#issuecomment-639398801" rel="noopener nofollow ugc">this guidance</a> which worked until today.</p>
<p>The relevant YAML code I’m using (full details are in the linked issue) is:</p>
<pre><code class="lang-yaml">    - name: Set up Node.js
      uses: actions/setup-node@v1
      with:
        node-version: '14'
    - name: Install Azurite storage emulator
      run: npm install -g azurite
    - name: Start Azurite blob endpoint
      shell: bash
      run: azurite-blob &amp;
</code></pre>
<p>Thanks!</p>

**Comment by jsoref** [comment, marked as answer]
<p>This isn’t a problem about action hosting, it’s a supply chain problem:</p><aside class="onebox githubissue" data-onebox-src="https://github.com/Marak/colors.js/issues/285">
  <header class="source">

      <a href="https://github.com/Marak/colors.js/issues/285" target="_blank" rel="noopener nofollow ugc">github.com/Marak/colors.js</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/Marak/colors.js/issues/285" target="_blank" rel="noopener nofollow ugc">Zalgo issue with `v1.4.44-liberty-2` release</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2022-01-08" data-time="04:26:12" data-timezone="UTC">04:26AM - 08 Jan 22 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/Marak" target="_blank" rel="noopener nofollow ugc">
          <img alt="Marak" src="https://user-images.githubusercontent.com/2119212/181092985-bd7bf72a-b3b4-42bb-ad27-8bec6ce071fc.png" class="onebox-avatar-inline" width="20" height="20">
          Marak
        </a>
      </div>
    </div>

    <div class="labels">
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">It's come to our attention that there is a zalgo bug in the `v1.4.44-liberty-2` <span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">release of colors.

Please know we are working right now to fix the situation and will have a resolution shortly.

![wolf-cola](https://user-images.githubusercontent.com/70011/148631268-d5b67a58-8794-4276-9bf7-e00b0faf1c27.jpeg)</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<aside class="onebox githubissue" data-onebox-src="https://github.com/Azure/Azurite/issues/1271#issuecomment-1008868400">
  <header class="source">

      <a href="https://github.com/Azure/Azurite/issues/1271#issuecomment-1008868400" target="_blank" rel="noopener nofollow ugc">github.com/Azure/Azurite</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/Azure/Azurite/issues/1271#issuecomment-1008868400" target="_blank" rel="noopener nofollow ugc">Azurite no longer runs correctly as a background task in GitHub Actions</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2022-01-09" data-time="23:43:36" data-timezone="UTC">11:43PM - 09 Jan 22 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/LarsKemmann" target="_blank" rel="noopener nofollow ugc">
          <img alt="LarsKemmann" src="https://user-images.githubusercontent.com/2119212/181092986-e23a116c-8a57-4741-81ab-45bf61124e80.jpeg" class="onebox-avatar-inline" width="20" height="20">
          LarsKemmann
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          question
        </span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">### Which service(blob, file, queue, table) does this issue concern?
Blob

##<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden"># Which version of the Azurite was used?
3.15.0

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)
NPM

### What's the Node.js version?
14.18.2

### What problem was encountered?
I've noticed that, since today, Azurite fails to start correctly in my GitHub Actions pipeline and instead dumps some unknown text/binary content to the GitHub Action log.

Action run before the issue started:
https://github.com/CareTogether/CareTogetherCMS/runs/4710008033?check_suite_focus=true#step:6:6

Action runs showing the issue:
https://github.com/CareTogether/CareTogetherCMS/runs/4755534824?check_suite_focus=true#step:6:6 (first attempt)
https://github.com/CareTogether/CareTogetherCMS/runs/4755601564?check_suite_focus=true#step:6:6 (second attempt)
https://github.com/CareTogether/CareTogetherCMS/runs/4755632933?check_suite_focus=true#step:6:6 (third attempt, reverting my .NET SDK upgrade to try to make the Action environment as identical to the previous successful run as possible)

### Steps to reproduce the issue?
See above. It seems like the following YAML is problematic now, but wasn't as of two days ago.
```yaml
    - name: Set up Node.js
      uses: actions/setup-node@v1
      with:
        node-version: '14'
    - name: Install Azurite storage emulator
      run: npm install -g azurite
    - name: Start Azurite blob endpoint
      shell: bash
      run: azurite-blob &amp;
```

### Have you found a mitigation/solution?
No. I can't figure out what is going on that might be affecting this. (On a related note, it would be helpful if there was some guidance provided by the Azurite GitHub readme/wiki for how to use Azurite in a GitHub Action "properly", if the above is improper somehow. I got this to work based on [guidance in an issue I found in the Azurite project](https://github.com/Azure/Azurite/issues/451#issuecomment-639398801) some time back provided by @ethomson.)</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Reply by jhendrix93** [reply]
Thank you. My name is Jordan Hendrix. .

**Comment by LarsKemmann** [comment]
<p>Yes - I figured that out about 10 minutes after posting this issue, but GitHub’s “akismet” bot had hidden my post as suspected spam. Hidden it even from me, apparently, so I couldn’t go back to update it. :man_facepalming:</p>
